src/js/features/data.js

```diff
@@ -0,0 +1,133 @@
+  overview: {},
+  chartData: [],
+  individuals: 0,
+  queryParameterStack: {},
+};
+
+const data = createSlice({
+  name: 'data',
+  initialState,
+  reducers: {
+    rearrange: (state, { payload }) => {
+      const chartsCopy = [...state.chartData];
+
+      let temp = payload.map((e) => chartsCopy.find((i) => e === i.name));
+      state.chartData = temp;
+      saveValue(LS_CHARTS_KEY, convertSequenceAndDisplayData(state.chartData));
+    },
+    disableChart: (state, { payload }) => {
+      state.chartData.find((e) => e.name === payload).isDisplayed = false;
+      saveValue(LS_CHARTS_KEY, convertSequenceAndDisplayData(state.chartData));
+    },
+    setDisplayedCharts: (state, { payload }) => {
+      state.chartData.forEach((val, ind, arr) => {
+        arr[ind].isDisplayed = payload.includes(val.name);
+      });
+      saveValue(LS_CHARTS_KEY, convertSequenceAndDisplayData(state.chartData));
+    },
+  },
+  extraReducers: {
+    [makeGetDataRequest.pending]: (state) => {
+      state.isFetchingData = true;
+    },
+    [makeGetDataRequest.fulfilled]: (state, { payload }) => {
+      const overview = payload.overview;
+      const queryParameterStack = payload.queryParameterStack;
+      const individuals = overview.individuals;
+
+      let fields = {
+        ...queryParameterStack,
+        ...queryParameterStack.extra_properties,
+      };
+      delete fields.extra_properties;
+
+      fields = Object.entries(fields).map((item) => ({
+        name: item[0],
+        data: item[1],
+        isExtraProperty: queryParameterStack.extra_properties.hasOwnProperty(
+          item[0]
+        ),
+      }));
+
+      const all_charts_obj = { ...overview, ...overview.extra_properties };
+      delete all_charts_obj.extra_properties;
+      delete all_charts_obj.individuals;
+
+      let all_charts = Object.entries(all_charts_obj).map((item) => ({
+        name: item[0],
+        data: item[1],
+        isDisplayed: true,
+      }));
+
+      all_charts.forEach((chart, index, arr) => {
+        arr[index].properties = fields.find((e) => e.name == chart.name)?.data;
+      });
+      all_charts.forEach((chart, index, arr) => {
+        arr[index].data = parseData(chart);
+      });
+
+      state.queryParameterStack = queryParameterStack;
+      state.overview = overview;
+      state.chartData = all_charts;
+      state.individuals = individuals;
+      state.isFetchingData = false;
+      state.fields = fields;
+
+      let convertedData = convertSequenceAndDisplayData(state.chartData);
+
+      const localValue = getValue(LS_CHARTS_KEY, convertedData, (val) =>
+        verifyData(val, convertedData)
+      );
+
+      const temp = [...state.chartData];
+
+      state.chartData = localValue.map((e) => ({
+        ...temp.find((v) => v.name === e.name),
+        isDisplayed: e.isDisplayed,
+      }));
+
+      convertedData = convertSequenceAndDisplayData(state.chartData);
+
+      saveValue(LS_CHARTS_KEY, convertedData);
```

Comment:
I'm not an expert on redux toolkit but this looks an awful lot like you have side effects in your reducers. These should be in action creators instead.

Reply:
I know but the otherway was to have an increased code complexity, which i wanted to kind of run away from. I would look into it.

Reply:
going through the code again , I realized that I could have done that in the thunk itself